driver/cri.go

```diff
@@ -0,0 +1,338 @@
+	start := time.Now()
+	cconfig := cconfigGlobal
+	pconfig := pconfigGlobal
+	cconfig.Metadata.Name = ctr.Name()
+	pconfig.Metadata.Name = defaultPodNamePrefix + cconfig.Metadata.Name
+
+	_, err := (*c.runtimeClient).CreateContainer(c.context, &pb.CreateContainerRequest{PodSandboxId: ctr.GetPodID(), Config: &cconfig, SandboxConfig: &pconfig})
+	if err != nil {
+		return "", 0, err
+	}
+	elapsed := time.Since(start)
+	msElapsed := int(elapsed.Nanoseconds() / 1000000)
+
+	return "", msElapsed, nil
+}
+
+// Stop will stop/kill a container
+func (c *CRIDriver) Stop(ctr Container) (string, int, error) {
+	start := time.Now()
+	resp, err := (*c.runtimeClient).ListContainers(c.context, &pb.ListContainersRequest{Filter: &pb.ContainerFilter{PodSandboxId: ctr.GetPodID()}})
+	if err != nil {
+		return "", 0, nil
+	}
+
+	containers := resp.GetContainers()
+	for _, ctr := range containers {
+		podID := ctr.GetPodSandboxId()
+		_, err := (*c.runtimeClient).StopContainer(c.context, &pb.StopContainerRequest{ContainerId: ctr.GetId(), Timeout: 0})
+		if err != nil {
+			log.Errorf("Error Stoping container %v", err)
+			return "", 0, nil
+		}
+		_, err = (*c.runtimeClient).StopPodSandbox(c.context, &pb.StopPodSandboxRequest{PodSandboxId: podID})
+		if err != nil {
+			log.Errorf("Error Stoping pod %v", err)
+			return "", 0, nil
+		}
+	}
+	elapsed := time.Since(start)
+	msElapsed := int(elapsed.Nanoseconds() / 1000000)
+
+	return "", msElapsed, nil
+}
+
+// Remove will remove a container
+func (c *CRIDriver) Remove(ctr Container) (string, int, error) {
+
+	start := time.Now()
+	resp, err := (*c.runtimeClient).ListContainers(c.context, &pb.ListContainersRequest{Filter: &pb.ContainerFilter{PodSandboxId: ctr.GetPodID()}})
+	if err != nil {
+		return "", 0, nil
+	}
+
+	containers := resp.GetContainers()
+	for _, ctr := range containers {
+		podID := ctr.GetPodSandboxId()
+		_, err = (*c.runtimeClient).RemoveContainer(c.context, &pb.RemoveContainerRequest{ContainerId: ctr.GetId()})
+		if err != nil {
+			log.Errorf("Error deleting container %v", err)
+			return "", 0, nil
+		}
+		_, err = (*c.runtimeClient).RemovePodSandbox(c.context, &pb.RemovePodSandboxRequest{PodSandboxId: podID})
+		if err != nil {
+			log.Errorf("Error deleting pod %v", err)
+			return "", 0, nil
+		}
+	}
+	elapsed := time.Since(start)
+	msElapsed := int(elapsed.Nanoseconds() / 1000000)
+
+	return "", msElapsed, nil
+}
+
+// Pause will pause a container
+func (c *CRIDriver) Pause(ctr Container) (string, int, error) {
+
+	return "", 0, nil
+}
+
+// Unpause will unpause/resume a container
+func (c *CRIDriver) Unpause(ctr Container) (string, int, error) {
+	return "", 0, nil
+}
+
+// Close allows the driver to free any resources/close any
+// connections
+func (c *CRIDriver) Close() error {
+	return nil
+}
+
+func openFile(path string) (*os.File, error) {
+	f, err := os.Open(path)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, fmt.Errorf("config at %s not found", path)
```

Comment:
minor, but since the function is `openFile` maybe the error shouldn't talk about configs; just "`file %s not found`"?